source/examples/xgboost-gpu-hpo-mnmg-parallel-k8s/notebook.ipynb

```diff
@@ -0,0 +1,727 @@
+{
```

Comment:
<p>It would cause a substantial overhead to launch a new cluster for each and every training job. If <code>n_trials=100</code>, we'd launch and tear down clusters 100 times. Shouldn't we prefer to keep the clusters until the entire hyperparameter optimization process is complete?</p>

Reply:
<p>That's true. I had though of <code>train_model</code> being called once per thread, but you're right it'll be called once per trial. Dask clusters spawned like this should be pretty lightweight, but I guess the overhead percentage depends how long each trial lasts.</p>